lib/rucio/core/replica.py

```diff
@@ -861,13 +868,14 @@ def _list_replicas(dataset_clause, file_clause, state_clause, show_pfns, schemes
 
             if 'scope' in file and 'name' in file:
                 if file['scope'] == scope and file['name'] == name:
-                    file['rses'][rse] += list(set([tmp_pfn[1] for tmp_pfn in pfns]))  # extract properly the pfn from the (domain, pfn)
+                    file['rses'][rse] += list(set([tmp_pfn[0] for tmp_pfn in pfns]))  # extract properly the pfn from the (pfn, domain, priority)
                     file['states'][rse] = str(state)
                     for tmp_pfn in pfns:
-                        file['pfns'][tmp_pfn[1]] = {'rse': rse,
+                        file['pfns'][tmp_pfn[0]] = {'rse': rse,
                                                     'type': str(rse_type),
                                                     'volatile': volatile,
-                                                    'domain': tmp_pfn[0]}  # extract properly the domain from the (domain, pfn)
+                                                    'domain': tmp_pfn[1],
+                                                    'priority': tmp_pfn[2]}
```

Comment:
I wonder if priority shouldn't be a total order. Because now you repeat 1-N for every RSE, but for the client it would be easier to have a total order I think.

Reply:
like in metalink?

Reply:
Yes basically. Because otherwise a client doesn't really know how to order the RSEs. Feels like we only get half the way we want to go in this case.

Reply:
It's correctly sorted though, but it will make Tobi's life easier I agree ;-D

Reply:
Is it? I thought all the PFNs are in one dictionary and the priority repeats 1-N?

Reply:
ah, now I see what you mean. yes, it's sorted per domain.